hazelcast/src/main/java/com/hazelcast/instance/BuildInfoProvider.java

```diff
@@ -63,6 +76,12 @@ public static BuildInfo getBuildInfo() {
         }
         int buildNumber = Integer.parseInt(build);
 
+        // override version with a system property
+        String overridingVersion = System.getProperty("hazelcast.version");
+        if (overridingVersion != null) {
+            version = overridingVersion;
+        }
+
```

Comment:
👍 for logging a message if used and keeping undocumented; having means to override the version is very useful for tests.